### PR TITLE
WIP: clap3.0.0-rc.7

### PR DESCRIPTION
**Pull request by conradludgate**
clap 3.0.0 is going into official release soon, with that comes the long awaited removal of structopt

Waiting on https://github.com/clap-rs/clap/issues/2869

### CODE REVIEW

**Comment by conradludgate** [comment]
Screenshot from 2021-12-10 17-20-07

Output compared from clap3 to the current structopt impl. Last test shows aliases still working

File: src/command/history.rs [hunk 1]
```diff
@@ -2,60 +2,55 @@ use std::env;
...
 use tabwriter::TabWriter;
 
 use atuin_client::database::Database;
 use atuin_client::history::History;
 use atuin_client::settings::Settings;
 use atuin_client::sync;
 
-#[derive(StructOpt)]
+#[derive(Subcommand)]
+#[clap(aliases = &["h", "hi", "his", "hist", "histo", "histor"])]
```
Comment (pksunkara) [inline src/command/history.rs:15]: Instead of all these aliases, `InferSubcommands` setting will provide the same result

Comment (conradludgate) [reply src/command/history.rs:15]: Thanks! I'll try it in the morning. Thanks for working on getting clap3 out there!

**Comment by conradludgate** [comment]
Small regression using `InferSubcommands` is that you cannot specify how to resolve ambiguous cases.

Originally, we supported `atuin history l` to be `atuin-history-list` and `atuin history la` to be `atuin-history-last`. The `InferSubcommands` setting just makes the base case an error

**Comment by pksunkara** [comment]
For those kind of conflict, alias will have to do. So add `l` as alias to `list`.

**Comment by conradludgate** [comment]
@ellie it's wholesome that my first PR here was using structopt instead of clap and very soon we'll be reverting that 😅 

https://github.com/ellie/atuin/pull/1

**Comment by ellie** [comment]
> @ellie it's wholesome that my first PR here https://github.com/ellie/atuin/pull/1 was using structopt instead of clap and very soon we'll be reverting that 😅 

Awww I love that haha! Come full circle 😁

**Comment by conradludgate** [comment]
Just rebased and updated to the lastest release candidate (rc7), as well as merge in support for shell completions (addressing #249)

**Comment by conradludgate** [comment]
Remind me to complete this

**Comment by panekj** [comment]
reminder that this can be closed due to https://github.com/ellie/atuin/pull/289

**Pull request closed without merging**